std::any errors

**Describe the bug**
The Conda build logs of PR #666 document that the current build with GNU 9.3.0 has problems using std::any:

```
bld/gqcp_1599474307839/work/gqcp/src/Basis/Integrals/IntegralEngine.cpp:18:
/home/runner/miniconda/envs/packaging/conda-bld/gqcp_1599474307839/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_p/include/libint2/util/any.h:34:12: error: 'std::any' has not been declared
   34 | using std::any; 
```

The 'standard' build/test and Docker build use GNU 7.5.0 and report no such error.

--- a/.github/workflows/docker-gqcp-build.yml
+++ b/.github/workflows/docker-gqcp-build.yml
@@ -1,7 +1,7 @@
 name: Docker build
 
 on:
-  push:
+  pull_request:
     paths:
       - 'Dockerfile'
       - '**.hpp'

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -7,7 +7,7 @@
 #   - the gqcpy Python bindings to the C++ library (optional)
 
 cmake_minimum_required(VERSION 3.13 FATAL_ERROR)
-project(GQCP VERSION 0.2.0 LANGUAGES CXX)
+project(GQCP VERSION 0.0.1 LANGUAGES CXX)
 list(APPEND CMAKE_MODULE_PATH ${CMAKE_SOURCE_DIR}/cmake)  # make CMake look into the ./cmake/ folder for configuration files
 
 

--- a/conda/build.sh
+++ b/conda/build.sh
@@ -1,5 +1,11 @@
 #!/usr/bin/env bash
 
+# Do not use C++17 standard
+# https://github.com/conda/conda-build/issues/3097
+if [[ ${target_platform} =~ .*linux.* ]]; then
+    CXXFLAGS="${CXXFLAGS//-std=c++17/}"
+fi
+
 mkdir build && cd build
 if [ `uname` == Darwin ]; then
     ${BUILD_PREFIX}/bin/cmake \

--- a/conda/conda_build_config.yaml
+++ b/conda/conda_build_config.yaml
@@ -6,6 +6,8 @@ c_compiler:
 cxx_compiler:
   - clangxx                       # [osx]
   - gxx                           # [linux]
+cxx_compiler_version:
+  - 7.5.0                         # [linux] 
 fortran_compiler:
   - gfortran
 target_platform:

--- a/conda/conda_build_config.yaml
+++ b/conda/conda_build_config.yaml
@@ -6,7 +6,9 @@ c_compiler:
 cxx_compiler:
   - clangxx                       # [osx]
   - gxx                           # [linux]
-cxx_compiler_version:
+c_compiler_version:               # [linux] 
+  - 7.5.0                         # [linux] 
+cxx_compiler_version:             # [linux] 
   - 7.5.0                         # [linux] 
 fortran_compiler:
   - gfortran

--- a/conda/meta.yaml
+++ b/conda/meta.yaml
@@ -42,7 +42,7 @@ requirements:
       - libint {{ libint }}
       - llvm-openmp               # [osx]
       - numpy
-      - intel_openmp              # [linux]
+      - intel-openmp              # [linux]
       - python
 test:
   commands:

--- a/gqcp/CMakeLists.txt
+++ b/gqcp/CMakeLists.txt
@@ -5,7 +5,7 @@ target_compile_options(gqcp PUBLIC -m64 -pipe)
 target_compile_options(gqcp PUBLIC "$<IF:$<CONFIG:Debug>,-g,-O2>")
 target_compile_options(gqcp PUBLIC "$<$<CXX_COMPILER_ID:GNU>:-pthread>")
 target_link_options(gqcp PUBLIC "$<$<CXX_COMPILER_ID:GNU>:-pthread>")
-target_compile_features(gqcp PUBLIC cxx_std_11)
+target_compile_features(gqcp PUBLIC cxx_std_17)
 
 if (OPTIMIZE_FOR_NATIVE)
     target_compile_options(gqcp PRIVATE "$<IF:$<CXX_COMPILER_ID:Intel>,-xHost,-march=native>")

--- a/gqcp/CMakeLists.txt
+++ b/gqcp/CMakeLists.txt
@@ -5,7 +5,7 @@ target_compile_options(gqcp PUBLIC -m64 -pipe)
 target_compile_options(gqcp PUBLIC "$<IF:$<CONFIG:Debug>,-g,-O2>")
 target_compile_options(gqcp PUBLIC "$<$<CXX_COMPILER_ID:GNU>:-pthread>")
 target_link_options(gqcp PUBLIC "$<$<CXX_COMPILER_ID:GNU>:-pthread>")
-target_compile_features(gqcp PUBLIC cxx_std_17)
+target_compile_features(gqcp PUBLIC cxx_std_14)
 
 if (OPTIMIZE_FOR_NATIVE)
     target_compile_options(gqcp PRIVATE "$<IF:$<CXX_COMPILER_ID:Intel>,-xHost,-march=native>")

--- a/gqcp/include/Basis/Integrals/BaseOneElectronIntegralBuffer.hpp
+++ b/gqcp/include/Basis/Integrals/BaseOneElectronIntegralBuffer.hpp
@@ -19,7 +19,6 @@
 
 
 #include "Mathematical/Representation/Matrix.hpp"
-
 #include <array>
 
 

